entgql/schema.go

```diff
@@ -24,12 +24,60 @@ import (
 	"github.com/vektah/gqlparser/v2/formatter"
 )
 
-type (
-	// TODO(giautm): refactor internal APIs
-	schemaGenerator struct {
-		graph     *gen.Graph
-		nodes     []*gen.Type
-		relaySpec bool
+// TODO(giautm): refactor internal APIs
+type schemaGenerator struct {
+	graph     *gen.Graph
+	nodes     []*gen.Type
+	relaySpec bool
+}
+
+var (
+	pos        = &ast.Position{Src: &ast.Source{BuiltIn: false}}
+	directives = map[string]*ast.DirectiveDefinition{
+		"goModel": {
+			Name:     "goModel",
+			Position: pos,
```

Comment:
position can be set to nil

Reply:
It's needed here, but not in other places. 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x150c2c2]

goroutine 1 [running]:
github.com/vektah/gqlparser/v2/formatter.(*formatter).FormatDirectiveDefinition(0x15a7f60, 0xc000347a40)
```

https://github.com/vektah/gqlparser/blob/master/formatter/formatter.go#L326-L328